extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -163,6 +163,16 @@ protected function getHTML( $title, $parserParams ) {
 			$timestamp = $latestRevision->getTimestamp();
 		} else {
 			$content = '';
+			$preloadTitle = Title::newFromText( urldecode( $parserParams['preload'] ) );
+
+			if ( isset( $preloadTitle ) && $preloadTitle->userCan( 'read' ) ) {
+				$rev = Revision::newFromTitle($preloadTitle);
+
+				if (is_object($rev)) {
+					$content = $this->parseWikitextFragment( $preloadTitle, $rev->getText() );
```

Comment:
This is POST so it does not take adventage of caching - consider calling here getHTML (itself) and passing $preloadTitle as a parameter - and then eventually overriding oldid and or timestamp in the response, e.g.:
```
$out = $this->getHtml( Title::newFromText( $parserParams['preload'] ) );
$out['oldid'] = 0;
return $out;
```
With some appropriate "checks" around.